lib/filewatch/watched_file.rb

```diff
@@ -0,0 +1,180 @@
+require "filewatch/buftok"
+
+module FileWatch
+  class WatchedFile
+    def self.new_initial(path, inode, stat = nil)
+      new(path, inode, stat, true)
+    end
+
+    def self.new_ongoing(path, inode, stat = nil)
+      new(path, inode, stat, false)
+    end
+
+    attr_reader :size, :inode, :state, :file, :buffer, :state_history
+    attr_reader :path, :filestat, :ignored_size, :accessed_at
+    attr_accessor :close_older, :ignore_older, :delimiter
+
+    def delimiter
+      @delimiter
+    end
+
+    def initialize(path, inode, stat, initial)
+      @path = path
+      @ignored_size = @size = 0
+      @inode = inode
+      @initial = initial
+      @state_history = []
+      @state = :watched
+      if (@filestat = stat).nil? && !@inode.nil?
+        restat
+      end
+      set_accessed_at
```

Comment:
The considerations of your comment are subtly interesting.

- The `set_accessed_at` at that point in the initializer is so that a call to `file_closable?` will never fail regardless of when it is called, but then it is only called once during the the actives processing at which point the file was opened and read. One edge case is a file that exists but is empty and remains so for longer than the close_older time will be 'closed'. This is OK and it will remain so until an append occurs saving unnecessary active processing and conserving file handles.

So I think it is OK to call `set_accessed_at` here.

- The class constructors `new_initial` and `new_ongoing` were designed to not require a stat upfront, however in practise, a stat is needed for the inode therefore inode and stat will never be nil, but the `initializer` protects against an unnecessary restat.
-  `@files = Hash.new { |h, k| h[k] = WatchedFile.new(k, nil, nil, false) }` could initialise a 'blank' watched_file.  Maybe this is too defensive, we will not get nil `@file` lookups but the only case where this might apply is if someone `unwatched` an arbitrary path but because of the default `if (watched_file = @files[path])` will always be true. Hmmm. Another argument to remove the  `unwatch` method and the Hash default and the `stat = nil` in the class constructors and the `if... restat` section. Also then we can do `private_class_method :new` to enforce the use of the class constructors.

Do you agree with the above code removals?